File: 0816/거리두기 확인/code.js
function solution(places) {
  let answer = Array.from({length:places.length}, ()=>0);
  
  places.forEach((place, idx)=>{
      let dx = [-1, 0, 1, 0];
      let dy = [0, 1, 0, -1];
      
      for(let i = 0; i<place.length; i++){
          for(let j = 0; j<place.length; j++){
              if(place[i][j]==="P"){
                  for(let k = 0; k<4; k++){
                      let nx = i + dx[k];
                      let ny = j + dy[k];
                      if(nx>=0 && nx<place.length && ny>=0 && ny<place.length && place[nx][ny]==="P"){
                          answer[idx]++
                      }
                  }
              }
              
              if(place[i][j]==="O"){
                  let cnt = 0
                  for(let k = 0; k<4; k++){
                      let nx = i + dx[k];
                      let ny = j + dy[k];
                      if(nx>=0 && nx<place.length && ny>=0 && ny<place.length && place[nx][ny]==="P"){
                          cnt++
                      }
                  }
                  
                  if(cnt > 1){
                      answer[idx]++
                  }
              }
          }
      }
  })
  answer = answer.map(v=>v>0?0:1)

  return answer;
}

File: 0830/두 큐 합 같게 만들기/code.py
from collections import deque

def solution(queue1, queue2):
    answer = -1
    cnt = 0;
    totalLeng = len(queue1) * 3
    queue1Leng = sum(queue1)
    queue2Leng = sum(queue2)
    _queue1 = deque(queue1)
    _queue2 = deque(queue2)
    
    while cnt < totalLeng:
        if queue1Leng < queue2Leng:
            ele = _queue2.popleft()
            queue1Leng += ele
            queue2Leng -= ele
            _queue1.append(ele)
            cnt += 1
        elif queue1Leng > queue2Leng:
            ele = _queue1.popleft()
            queue2Leng += ele
            queue1Leng -= ele
            _queue2.append(ele)
            cnt += 1
        else:
            return cnt
    
    return answer